ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/TrailingCommaRuleTest.kt

```diff
@@ -1021,4 +1021,157 @@ class TrailingCommaRuleTest {
             ).isFormattedAs(formattedCode)
     }
+
+    @Test
+    fun `Given that a trailing comma is required then add trailing comma after last enum member`() {
+        val code =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE
+            }
+            """.trimIndent()
+        val formattedCode =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE,
+            }
+            """.trimIndent()
+
+        trailingCommaRuleAssertThat(code)
+            .withEditorConfigOverride(allowTrailingCommaProperty to true)
+            .hasLintViolation(3, 13, "Missing trailing comma before \"}\"")
+            .isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Given that a trailing comma is is not allowed then remove comma after last enum member`() {
+        val code =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE,
+            }
+            """.trimIndent()
+        val formattedCode =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE
+            }
+            """.trimIndent()
+
+        trailingCommaRuleAssertThat(code)
+            .withEditorConfigOverride(allowTrailingCommaProperty to false)
+            .hasLintViolation(3, 13, "Unnecessary trailing comma before \"}\"")
+            .isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Given that a trailing comma is not allowed then nothing happens for enums terminated with semicolon`() {
+        val code =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE;
+
+                fun print() = name()
+            }
+            """.trimIndent()
+
+        trailingCommaRuleAssertThat(code)
+            .withEditorConfigOverride(allowTrailingCommaProperty to false)
+            .hasNoLintViolations()
+    }
+
+    @Test
+    fun `Given that a trailing comma is allowed then nothing happens for enums terminated with semicolon`() {
+        val code =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE;
+
+                fun print() = name()
+            }
+            """.trimIndent()
+
+        trailingCommaRuleAssertThat(code)
+            .withEditorConfigOverride(allowTrailingCommaProperty to true)
+            .hasNoLintViolations()
+    }
+
+    @Test
+    fun `Given that a trailing comma is allowed then it is added for complicated enums`() {
+        val code =
+            """
+            interface Printable {
+                fun print(): String
+            }
+
+            enum class Shape : Printable {
+                /*
+                 * Block comment is OK!
+                 */
+                Square {
+                    override fun print() = "■" // EOL Comment should be OK
+                },
```

Comment:
Please split this test. The test cases with comments deserve their own test. Also this simplifies this example a bit.

Reply:
Agreed, will do